### PR TITLE
Endpoint composition xml update and loader enhancements for Conformance and Constraints

### PR DESCRIPTION
**Pull request by paulr34**
Endpoint Composition XML Update and Loader Enhancements for Conformance and Constraints

- Added Conformance and Constraint to the Device Composition database table.
- Updated comments for better clarity.
- Added try/catch blocks for error handling.
- General code cleanup.
- Updated schema to avoid confusion with the SQL keyword CONSTRAINT.
- Update schema diagram
- Adding prettier formatting

Updates to XML based on this PR https://github.com/project-chip/connectedhomeip/pull/34818

### CODE REVIEW

File: docs/zap-schema.svg
Comment (brdandu) [inline docs/zap-schema.svg:1]: All the unnecessary changes might be related to your npm packages not being unto date. Run npm install and then run `prettier -w ./` once so that you can revert the formatting

Comment (paulr34) [reply docs/zap-schema.svg:1]: Interesting, thank you!

**Review by ethanzhouyc — APPROVED** [review]

File: src-electron/db/query-loader.js [hunk 1]
```diff
@@ -938,54 +938,80 @@ async function insertAtomics(db, packageId, data) {
...
-  } else {
-    return dbApi.dbInsert(
-      db,
-      'INSERT INTO ENDPOINT_COMPOSITION (TYPE, CODE) VALUES (?, ?)',
-      [composition.compositionType, composition.code],
-    )
+async function insertEndpointComposition(db, composition, context) {
+  try {
+    if (parseInt(context.mandatoryDeviceTypes, 16) === composition.code) {
+      return await dbApi.dbInsert(
```
Comment (tecimovic) [inline src-electron/db/query-loader.js:944]: You do not do `return await xxxx` if this is an async function. You just retun the async itself.

File: src-electron/db/query-loader.js [hunk 2]
```diff
@@ -938,54 +938,80 @@ async function insertAtomics(db, packageId, data) {
...
+        db,
+        'INSERT INTO ENDPOINT_COMPOSITION (TYPE, CODE) VALUES (?, ?)',
+        [dbEnum.composition.mandatoryEndpoint, composition.code],
+      )
+    } else {
+      return await dbApi.dbInsert(
```
Comment (tecimovic) [inline src-electron/db/query-loader.js:950]: Same thing.

File: src-electron/db/query-loader.js [hunk 3]
```diff
@@ -938,54 +938,80 @@ async function insertAtomics(db, packageId, data) {
...
+        db,
+        'INSERT INTO ENDPOINT_COMPOSITION (TYPE, CODE) VALUES (?, ?)',
+        [composition.compositionType, composition.code],
+      )
+    }
+  } catch (error) {
+    console.error('Error inserting endpoint composition:', error)
+    throw error // Re-throw the error after logging it
```
Comment (tecimovic) [inline src-electron/db/query-loader.js:958]: These seem needless. You're already getting an error. You are also not "logging" it. You're just printing it out.
We have our "log" module if you want to log stuff.

File: src-electron/db/query-loader.js [hunk 4]
```diff
@@ -938,54 +938,80 @@ async function insertAtomics(db, packageId, data) {
...
   const query =
     'SELECT ENDPOINT_COMPOSITION_ID FROM ENDPOINT_COMPOSITION WHERE CODE = ?'
-  const result = await dbApi.dbGet(db, query, [deviceType.code])
-  return result ? result.ENDPOINT_COMPOSITION_ID : null
+  try {
+    const result = await dbApi.dbGet(db, query, [deviceType.code])
+    return result ? result.ENDPOINT_COMPOSITION_ID : null
+  } catch (error) {
+    console.error('Error retrieving endpoint composition ID:', error)
+    return null
```
Comment (tecimovic) [inline src-electron/db/query-loader.js:980]: ?
This is again simply eating an error and printing it on the console, which is mostly lost.
You need to use the "log" module.

File: src-electron/db/query-loader.js [hunk 5]
```diff
@@ -938,54 +938,80 @@ async function insertAtomics(db, packageId, data) {
...
+  try {
+    return dbApi.dbInsert(db, insertQuery, [
+      parseInt(deviceType.childDeviceId, 16),
+      endpointCompositionId,
+      deviceType.conformance,
+      deviceType.constraint,
+    ])
+  } catch (error) {
+    console.error('Error inserting device composition:', error)
+    throw error // Re-throw the error after logging it
```
Comment (tecimovic) [inline src-electron/db/query-loader.js:1013]: ? More of this....

File: src-electron/rest/user-data.js [hunk 6]
```diff
@@ -50,7 +50,7 @@ function httpGetSessionKeyValues(db) {
     let sessionId = request.zapSessionId
     let sessionKeyValues = await querySession.getAllSessionKeyValues(
       db,
-      sessionId
+      sessionId,
```
Comment (tecimovic) [inline src-electron/rest/user-data.js:53]: What is the comma doing here?

File: src-electron/rest/user-data.js [hunk 7]
```diff
@@ -83,7 +83,7 @@ function httpGetSessionNotifications(db) {
     let sessionId = request.zapSessionId
     let notifications = await querySessionNotification.getNotification(
       db,
-      sessionId
+      sessionId,
```
Comment (tecimovic) [inline src-electron/rest/user-data.js:86]: ?

File: src-electron/rest/user-data.js [hunk 8]
```diff
@@ -212,7 +212,7 @@ function httpPostCluster(db) {
         .getSessionPackagesByType(
           db,
           sessionId,
-          dbEnum.packageType.zclProperties
+          dbEnum.packageType.zclProperties,
```
Comment (tecimovic) [inline src-electron/rest/user-data.js:215]: ? This is a function call.

File: src-electron/rest/user-data.js [hunk 9]
```diff
@@ -330,17 +330,17 @@ function httpPostAttributeUpdate(db) {
           paramArray,
           reportMinInterval,
           reportMaxInterval,
-          reportableChange
-        )
-      )
+          reportableChange,
+        ),
```
Comment (tecimovic) [inline src-electron/rest/user-data.js:334]: ?

**Review by tecimovic — COMMENTED** [review]
THere are many issues with this PR....

File: src-electron/db/query-loader.js (hunk 1 above)
Comment (paulr34) [reply src-electron/db/query-loader.js:944]: Ok I'll fix that

File: src-electron/db/query-loader.js (hunk 3 above)
Comment (paulr34) [reply src-electron/db/query-loader.js:958]: Ok is this an issue?

File: src-electron/rest/user-data.js (hunk 6 above)
Comment (paulr34) [reply src-electron/rest/user-data.js:53]: This was a problem with the formatter changes that is being fixed with this PR..

File: src-electron/rest/user-data.js (hunk 8 above)
Comment (paulr34) [reply src-electron/rest/user-data.js:215]: Again, formatter issue

**Comment by paulr34** [comment]
> THere are many issues with this PR....

Anything besides your comments? Because those were not issues

File: src-electron/db/query-loader.js (hunk 3 above)
Comment (tecimovic) [reply src-electron/db/query-loader.js:958]: Yes. Put it in the log. We don't want to use `console.error`
